sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -2531,7 +2548,24 @@ class RapidsConf(conf: Map[String, String]) extends Logging {
 
   lazy val chunkedReaderEnabled: Boolean = get(CHUNKED_READER)
 
-  lazy val chunkedSubPageReaderEnabled: Boolean = get(CHUNKED_SUBPAGE_READER)
+  lazy val limitChunkedReaderMemoryUsage: Boolean = {
+    val hasLimit = get(LIMIT_CHUNKED_READER_MEMORY_USAGE)
+    val deprecatedConf = get(CHUNKED_SUBPAGE_READER)
+
+    if(deprecatedConf.isDefined) {
+      logWarning(s"'${CHUNKED_SUBPAGE_READER.key}' is deprecated and is replaced by " +
+        s"'${LIMIT_CHUNKED_READER_MEMORY_USAGE}'.")
+      if(hasLimit.isDefined && hasLimit.get != deprecatedConf.get) {
+        throw new IllegalStateException(s"Both '${CHUNKED_SUBPAGE_READER.key}' and " +
+          s"'${LIMIT_CHUNKED_READER_MEMORY_USAGE.key}' are set but using different values.")
+      }
+      deprecatedConf.get
+    } else {
+      hasLimit.getOrElse(true)
+    }
```

Comment:
I think  we should
- Give precedence to the new conf if defined. 
- WARN if the deprecated conf is defined too, that it is being ignored
- Never throw

Reply:
I don't think we should select one and ignore the other, similar to the other configs in Spark (like `spark.sql.legacy.parquet.int96RebaseModeInRead`). As such, here I always make sure if both configs were set then they must have the same value, otherwise will just parse the only config that has been set.

Reply:
Probably better to check @felixcheung what the policy should be to avoid inconsistencies in our project. Another example of deprecation does not do anything drastic https://github.com/NVIDIA/spark-rapids/blob/66f2cc528df0c9afa89785d10079432caf274212/sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala#L2602-L2617

Reply:
IMO if we decide to ignore the old config, it's better to throw instead of just printing a warning then proceed. The users may not be aware of such warnings at all and receive a wrong outcome.

Reply:
Your suggestion definitely has merit and should be considered.  However as a user I would prefer an imperfect rule that can applied and easily remembered than explain a piecemeal of exceptions.

Reply:
I agree with @gerashegalov here.

Reply:
Thanks all. I've changed the logic of reading conf as we discussed:
 * Prioritize reading value from the new conf
 * Warn if the deprecated conf is set
 * Throw if both confs are set but with different values

Reply:
What is the point of prioritizing if you end up looking at both if both are set, just so you can throw the exception?

Reply:
Your point makes sense. That's because the priority here is just redundant: we end up checking both configs anyway, to make sure there will be no surprise in the output.

Anyway, I've filed an issue to remove the deprecated config: https://github.com/NVIDIA/spark-rapids/issues/10735